### DISCUSSION TITLE
Can rocket add qwik's resumability like feature?

### DISCUSSION THREAD

**Discussion by blueray453** [rwf2/Rocket#2308] [General]
Can rocket add qwik's resumability like feature?

https://github.com/BuilderIO/qwik

Best Explained in:
https://www.youtube.com/watch?v=lY6e7Hw4uVo
https://www.youtube.com/watch?v=_PDpoJUacuc
https://www.youtube.com/watch?v=0dC11DMR3fU

Blogs that can be interesting:
https://www.builder.io/blog/dont-blame-the-developer-for-what-the-frameworks-did
https://www.builder.io/blog/hydration-is-pure-overhead
https://www.builder.io/blog/why-progressive-hydration-is-harder-than-you-think
https://www.builder.io/blog/from-static-to-interactive-why-resumability-is-the-best-alternative-to-hydration
https://www.builder.io/blog/why-is-builderio-creating-qwik-and-partytown
https://www.builder.io/blog/how-we-cut-99-percent-js-with-qwik-and-partytown
https://www.builder.io/blog/our-current-frameworks-are-on-we-need-o1
https://www.builder.io/blog/death-by-closure
https://www.builder.io/blog/introducing-qwik-framework
https://www.builder.io/blog/introducing-qwik-starters

The optimizer (https://github.com/BuilderIO/qwik/tree/main/packages/qwik/src/optimizer) is written in rust. It mainly handle the $ at the end.

**Comment by the10thWiz** [comment]
Probably not, for a few different reasons. First, Rocket is a fairly minimal web framework, and doesn't include many advanced features that frameworks like qwik have. Rocket is purely server-side.

There are also some technical issue to handle in the process of making something like this work. First, a tighter integration between some Rust compiled for Wasm & natively to enable both the server and client to execute the same code share state. This requires a component model (which Rocket does not natively have), some concept of state w/ serialization, as well as some ability to resume execution anywhere.

This feature can, and should be implemented outside of Rocket. Rocket already provides the necessary functionality to make this work, so no changes to Rocket are needed. I suspect Yew is a good target to work on this, but it would need to be ported to native Rust, and generate Html.

**Comment by blueray453** [comment]
Thank you very much. I was confused whether this needs to be implemented on rocket or yew. As the optimizer is already written in rust, i think this would be easier. However, it would be better if resumability was a framework agnostic library. That way we could use it on any framework.

**Reply by the10thWiz** [reply]
Resumability can't really be implemented in a framework agnostic way. Arguably, Hydration or any similar alternative can't really be implemented in that way either, especially in Rust. JS, since it's an interpreted language, can be paused in a much easier way, but even then, frameworks have to implement their own resumability / hydration.
